### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by WilsonBradley** [curif/AgeOfJoy-2022.1#63] [General]
Are the Arcade cabinets dynamic or static in rooms? 
Means, when starting up, are the Roms assigned a specific cabinet and location or is it dynamic? Just loads up all cabinets first in first out?

**Comment by curif** [comment]
When you copy the cabinets assets to the `/cabinets` path, the game will decompress them to `/cabinetsdb` (when you start the game).

Later, when you enter each room, an algorithm will analyze the slots available, and assign the unassigned new cabinets to that room, replacing the 'out of order' cabinets with the new ones. Also, it will register the position so the next time you enter that room, you can find the cabinet in the same place.

You can "hack" the game, changing the positions by modifying the `/cabinetsdb/registry.json`

For example, to replace frogger with dkong, change:

`{"CabinetDBName":"dkong","Rom":"dkong.zip","Room":"Room001","Position":1},{"CabinetDBName":"frogger","Rom":"frogger.zip","Room":"Room001","Position":2}`

for

`{"CabinetDBName":"frogger","Rom":"dkong.zip","Room":"Room001","Position":1},{"CabinetDBName":"dkong","Rom":"frogger.zip","Room":"Room001","Position":2}`

**Comment by yepyep3** [comment]
Awesome app! Defender starts & I see coin credits, but I can't start the game - any ideas on what could be wrong?

**Reply by curif** [reply]
Did you check the md5 of you rom?

**Reply by WilsonBradley** [reply]
> Awesome app! Defender starts & I see coin credits, but I can't start the game - any ideas on what could be wrong?
https://youtu.be/ULjvFlGZ_lY

https://curif.notion.site/AGE-of-Joy-Rom-compatibility-Database-a3793cebeb5c4a7f9b1cd2c3306d2ee1

**Reply by WilsonBradley** [reply]
Using 0.1.0
Also having problems starting games now. Putting in quarters gives sound but credits don't show.
All games

*UPDATE: 
After searching/finding the MAME_2003-Plus package, all ROMs were compatible and machines worked.
https://archive.org/details/MAME_2003-Plus_Reference_Set_2018

**Reply by yepyep3** [reply]
Thanks - any chance you can post a link to the proper mame or a good resource for them?

**Reply by curif** [reply]
Sorry, posting anything related to MAME roms is against the rules. The Romsets must be compatible with Libretro-mame 2003 plus. You can search for the  MAME 2003-Plus Reference: Full Non-Merged Romsets in Google.

I some games you need to put a second coin to start the game, after the game starting phase.

**Reply by yepyep3** [reply]
Ok - thx very much for your great work on this & responsiveness!

**Reply by curif** [reply]
You can check the MD5 sum with this [tool](https://curif.github.io/AgeOfJoy-ROMCRC/index.html)
